frontend/src/utilsLogic.tsx

```diff
@@ -192,7 +213,9 @@ export const joinInstancesAndTemplates = (
 ) =>
   templates.map(t => ({
     ...t,
-    instances: instances.filter(i => i.idTemplate === t.id),
+    instances: instances.filter(
+      i => i.templateId === makeTemplateKey(t.id, t.workspaceId!)
```

Comment:
Also here, it's better to accept _undefinable_ parameters and ensuring they're undefined.

Reply:
I solved it avoiding undefined value in makeTemplateKey function